### DISCUSSION TITLE
Can I generate a static website on Windows?

### DISCUSSION THREAD

**Discussion by Zauberweib** [datenstrom/community#115] [Ask a question] [answered]
I want to try to convert the content to static page using the commandbox via xampp-control, writing: `php yellow.php build /test`. Many files were created but the textfiles don't. they were named without any ending; in the commandwindow I get the message: "can't write file..." 
the index.htmls were done but none of these created by me (via browser-edit) 
whats gone wrong?

**Comment by markseuffert** [comment]
Which Yellow version are you using? Can you copy/paste the output of the [command line](https://github.com/annaesvensson/yellow-command)? Thanks.

**Comment by Zauberweib** [comment]
its the actual version 0.6.1. 

it says:

# php yellow.php build /test

ERROR building content location '/search/', HTTP/1.1 500 Server error: Static we
bsite not supported!
ERROR building content location '/blog/published:2015-10/', HTTP/1.1 500 Server
error: Can't write file '/test/blog/published:2015-10/index.html'!
ERROR building content location '/blog/published:2013-04/', HTTP/1.1 500 Server
error: Can't write file '/test/blog/published:2013-04/index.html'!
ERROR building content location '/blog/author:schnurpsel/', HTTP/1.1 500 Server
error: Can't write file '/test/blog/author:schnurpsel/index.html'!
ERROR building content location '/blog/author:zaubsie/', HTTP/1.1 500 Server err
or: Can't write file '/test/blog/author:zaubsie/index.html'!
ERROR building content location '/blog/author:yellow/', HTTP/1.1 500 Server erro
r: Can't write file '/test/blog/author:yellow/index.html'!
ERROR building content location '/blog/tag:example/', HTTP/1.1 500 Server error:
 Can't write file '/test/blog/tag:example/index.html'!
ERROR building content location '/blog/tag:beispiel/', HTTP/1.1 500 Server error
: Can't write file '/test/blog/tag:beispiel/index.html'!
ERROR building content location '/blog/tag:blog/', HTTP/1.1 500 Server error: Ca
n't write file '/test/blog/tag:blog/index.html'!
ERROR building content location '/blog/tag:archiv/', HTTP/1.1 500 Server error:
Can't write file '/test/blog/tag:archiv/index.html'!
ERROR building content location '/blog/tag:links/', HTTP/1.1 500 Server error: C
an't write file '/test/blog/tag:links/index.html'!
ERROR building content location '/blog/tag:datei/', HTTP/1.1 500 Server error: C
an't write file '/test/blog/tag:datei/index.html'!
ERROR building content location '/blog/tag:/', HTTP/1.1 500 Server error: Can't
write file '/test/blog/tag:/index.html'!
ERROR building content location '/blog/tag:autoren/', HTTP/1.1 500 Server error:
 Can't write file '/test/blog/tag:autoren/index.html'!
Yellow build: 25 content, 4 media, 5 system, 14 errors, status 500

thanks by now!

**Comment by nibreh** [comment]
Hello. 
Don't know what's wrong, but my first reflex in that case is to check permissions on files & folders.

**Comment by markseuffert** [comment]
Looks like you are using Windows, where creating static files doesn't work properly.

**Comment by Zauberweib** [comment]
yes, I'm using Windows. permissions are ok. The files were made, but they are named e.g. "neue-seite" instead of "neue-seite.html". Also the pathes in links are not correct.

**Comment by markseuffert** [comment]
I am sorry to say, the filesystem on [Windows does not support files with a colon character](https://msdn.microsoft.com/en-us/library/windows/desktop/aa365247%28v=vs.85%29.aspx). Because the colon character is used by the Windows filesystem, you can not access all links in your blog. Your web server probably shows an error 403 on the blog pages you mentioned above.

How to fix this? Use a Mac or Linux. They probably provide the best working environment for web developers. If anyone knows a solution for Windows, please let us know. Any help is appreciated.

**Comment by Zauberweib** [comment]
oh, ok. "again what learned" would Loddar say :)  thanks for you answer.

**Comment by hysonlee** [comment]
I also noticed this problem, build under a non-windows all is fine.
Root cause is the colon mark(:), so this issue should be closed?

**Comment by markseuffert** [comment]
See also datenstrom/yellow#125

**Comment by markseuffert** [comment, marked as answer]
Should be working now, Windows support was added to latest Yellow.

**Comment by Zauberweib** [comment]
Thanks. It does now anything, but there were no link working. Maybe it's up to me, was only some quick-fast-hurry-hurry-let's-try-it ... no wonder. I'll try again calmly on weekend. 
But I want to say thanks for stay tuned on this problem.

**Comment by Zauberweib** [comment]
Ok, I tried it again. "Error building location /search/ - Static website not supported"
single sites have no ending (but opening in browser works). 
the connection to css doesn't work. wrong link header of each site. 

sorry for having no better news...

**Comment by markseuffert** [comment]
Sorry, the search is not supported in a static website.
The rest is probably something with your configuration, please open a new issue.

**Comment by Zauberweib** [comment]
ok, thanks.
